app/src/androidTest/java/com/github/se/cyrcle/ui/map/MapScreenTest.kt

```diff
@@ -186,12 +160,94 @@ class MapScreenTest {
     userViewModel.setCurrentUser(TestInstancesUser.user1)
 
     composeTestRule.setContent {
-      MapScreen(navigationActions, parkingViewModel, userViewModel, mapViewModel, permissionHandler)
+      MapScreen(
+          navigationActions,
+          parkingViewModel,
+          userViewModel,
+          mapViewModel,
+          permissionHandler,
+          addressViewModel)
     }
     composeTestRule.waitUntilExactlyOneExists(hasTestTag("addButton"))
     // Perform click on the add button
     composeTestRule.onNodeWithTag("addButton").performClick()
 
     verify(navigationActions).navigateTo(Route.ADD_SPOTS)
   }
+
+  /**
+   * Test to verify that settings menu is displayed and advanced mode switch is displayed when
+   * settings button is clicked.
+   */
+  @Test
+  fun testSettingsMenu() {
+    composeTestRule.setContent {
+      MapScreen(
+          mockNavigation,
+          parkingViewModel,
+          userViewModel,
+          mapViewModel,
+          permissionHandler,
+          addressViewModel)
+    }
+
+    composeTestRule.onNodeWithTag("SettingsMenuButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("SettingsMenuButton").performClick()
```

Comment:
I think that it is common practice to call `isClickable` before calling `performClick`, as this ensure that there is some action do perform.

Reply:
Done thanks !